File: veins_test/cpp/stdin_upper.cpp
#!/usr/bin/env -S fragletc --vein=cpp
#include <iostream>
#include <cctype>
int main() {
    char c;
    while (std::cin.get(c)) std::cout << static_cast<char>(std::toupper(static_cast<unsigned char>(c)));
    return 0;
}

File: veins_test/zig/stdin_upper.zig
#!/usr/bin/env -S fragletc --vein=zig
const std = @import("std");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();
    const input = try std.io.getStdIn().readToEndAlloc(allocator, 1024 * 1024);
    defer allocator.free(input);
    for (input) |c| {
        const out = [_]u8{std.ascii.toUpper(c)};
        _ = try std.io.getStdOut().writer().write(&out);
    }
}
